File: java-webflux/src/main/java/com/ederfmatos/javawebflux/repository/TransactionRepository.java
package com.ederfmatos.javawebflux.repository;

import com.ederfmatos.javawebflux.entity.Transaction;
import reactor.core.publisher.Mono;

public interface TransactionRepository {
    Mono<Void> create(Transaction transaction);
    Mono<Transaction> findById(String id);
}

File: kotlin-webflux/src/main/kotlin/com/ederfmatos/kotlinwebflux/repository/TransactionRepository.kt
package com.ederfmatos.kotlinwebflux.repository

import com.ederfmatos.kotlinwebflux.entity.Transaction
import reactor.core.publisher.Mono

interface TransactionRepository {
    fun create(transaction: Transaction): Mono<Void>
    fun findById(id: String): Mono<Transaction>
}
